src/main/java/duke/Task.java

```diff
@@ -0,0 +1,316 @@
+            } catch (DateTimeParseException e) {
+                throw new DukeException.DateInputInvalidException();
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "[" + Type.DEADLINE.getSymbol() + "]" + super.toString()
+                    + " (by: " + dateInputFormatter.format(time) + ")";
+        }
+    }
+
+    /** A type of duke.Task that starts at a specific time and ends at a specific date/time. */
+    public static class Event extends Task {
+        private final LocalDate time;
+
+        public Event(String description, String time) throws DukeException {
+            super(description);
+            try {
+                this.time = LocalDate.parse(time, dateInputFormatter);
+            } catch (DateTimeParseException e) {
+                throw new DukeException.DateInputInvalidException();
+            }
+        }
+
+        public Event(String description, boolean isDone, String time) throws DukeException {
+            super(description, isDone);
+            try {
+                this.time = LocalDate.parse(time, dateInputFormatter);
+            } catch (DateTimeParseException e) {
+                throw new DukeException.DateInputInvalidException();
+            }
+        }
+
+        @Override
+        public String toString() {
+            return "[" + Type.EVENT.getSymbol() + "]" + super.toString()
+                    + " (at: " + dateInputFormatter.format(time) + ")";
+        }
+    }
+
+    /**
+     * Parses the data from the file to return a duke.Task.
+     *
+     * @param str The String input read from the file.
+     * @return The duke.Task whose data is stored in the file.
+     * @throws DukeException The exception thrown when the file data is invalid.
+     */
+    public static Task readTaskFromFile(String str) throws DukeException {
+        String regex = "^\\[([TDE])]\\[([X ])] (.+)$";
+
+        // Create a Pattern object
+        Pattern pattern = Pattern.compile(regex);
+        Matcher m = pattern.matcher(str);
+
+        String taskSymbol;
+        String statusIcon;
+        String details;
+
+        // check if file string matches the format
+        if (m.find()) {
+            taskSymbol = m.group(1);
+            statusIcon =  m.group(2);
+            details = m.group(3);
+        } else {
+            throw new DukeException.FileDataInvalidException();
+        }
+
+        boolean isDone;
+
+        // parse task type
+        Type type = Type.getTypeFromSymbol(taskSymbol);
+        if (type == null) {
+            throw new DukeException.FileDataInvalidException();
+        }
+
+        // parse status icon
+        if (statusIcon.equals(DONE_STATUS_ICON)) {
+            isDone = true;
+        } else if (statusIcon.equals(NOT_DONE_STATUS_ICON)) {
+            isDone = false;
+        } else {
+            throw new DukeException.FileDataInvalidException();
+        }
+
+        switch (type) {
+            case TODO:
```

Comment:
Should the case indentation be 4 spaces rather than 8 for our java coding standard? 💭